### DISCUSSION TITLE
Custom colors in my tailwind.config.js file not showing on page

### DISCUSSION THREAD

**Discussion by spiercy16** [tailwindlabs/tailwindcss#2016] [Help]
I just installed Tailwind in VS Code (noob to Tailwind) and am having a problem where the color I configured in tailwind.config.js does not work on my page. When I add the Tailwind code to my class in VS Code I'm able to see the value (bg-bistre) with the actual color tile displayed next to, so it's somehow referencing the page, but when I view the page in the browser the color does not work. If I change the value to a default Tailwind value however it does work (like bg-black-500). 

Here is my code, though it's almost identical to the code found in the official tailwind tutorial [here](https://tailwindcss.com/course/customizing-your-design-system) : 

module.exports = {
  purge: [],
  theme: {
    extend: {
		colors: {
			'bistre': '#231006',
			'dutch-white': '#EFE4C5',
			'kombu-green': '#243010',
			'polished-pine': '#539987',
			'midnight-green': '#1A535C',
		},
	}
  },
  variants: {},
  plugins: [],
}

One discrepancy I noticed is that in the Tailwind tutorial video they used single quotes around the color noames and values, where as in their documentation on configuring a theme they omitted them (confusing for new users, especially if they're tyring to troubleshoot issues to have conflicting code). I used them because I get an error message in VS Code otherwise. 

Thanks in advance for any help. I've already wasted about an hour trying to troubleshoot it already.

**Comment by estevanmaito** [comment]
Did you recompile your CSS after you've made these changes?

Something like this: https://tailwindcss.com/docs/installation#using-tailwind-cli

**Reply by spiercy16** [reply]
I got it working, thanks!

**Reply by chrisonedata** [reply]
@estevanmaito Please help. How do you compile in Laravel? I already tried npm run dev and npm run watch but the custom color is not reflecting.

**Reply by daniel-norris** [reply]
@chrisonedata 
 
You need to follow the same instructions in the link above and also follow the "Laravel Mix" instructions to get it working with Laravel. 

https://tailwindcss.com/docs/installation#using-tailwind-cli

**Reply by Ekeu** [reply]
Hello,

Please I having the same issue trying to use custom colors in tail wind but with no avail.

I'm actually using Tail wind with Nuxt and I followed the steps given by the docs but up till now I can't still make it work.

Here is the basic config I did :

```
module.export = {
  purge: [],
  theme: {
    extend: {
      colors: {
        'nebula': "#5468ff"
      }
    }
  },
  variants: {},
  plugins: []
};
```
When using it for example on this hr tag:

`<hr class=" w-12 h-1 bg-nebula rounded-full mt-8" />`

It won't work.

After setting up the config I ran again` npm run dev` to recompile the css but it isn't working.

Please Help

**Reply by simonswiss** [reply]
hey @Ekeu - could you please share your project GitHub repo (or a reproduction of the state you are in, so we can help troubleshoot)?

**Reply by Ekeu** [reply]
Hey @simonswiss  

Thanks for the feedback. When you say state you mean my current config ??

**Reply by shiftgeist** [reply]
@Ekeu I think what he means is a **minimal reproduction** (GitHub repo) that shows what environment the tailwind config is being used in.

**Reply by Ekeu** [reply]
Hey I just want to thank you for your feedback.

I was actually able to make it run. I'm not pretty sure if that is the reason but my first `tailwind.config.js` file I created it myself by creating a new file but later after several attempts I deleted it and went back to tailwind's doc and used the command they provided which generated a new `tailwind.config.js` and it worked.

So I concluded saying I had to instead use the command to create my config file rather than creating it myself.

**Reply by mabitten** [reply]
@Ekeu  Do you can to share with us that comand?

**Reply by Helpico** [reply]
npx tailwind init

**Reply by simonswiss** [reply]
Ok I know this thread is 1 year old and likely not really helpful to respond now, but... reading this again today I saw you had `module.export =` in your code shared here from the manual config file you made:

![image](https://user-images.githubusercontent.com/485747/136915520-864307e7-2e0f-490f-9d56-ad9de32849c4.png)

And this should be `module.exports` with an `s` - that was possibly the issue 👍

**Comment by homocodian** [comment]
Just move the colors property under theme property, remove from extend property.
Then you will only see your defined colors by tailwind css intellisense.

**Reply by Azalea9X** [reply]
This works, though, I noticed when I compile the tailwind css it adds in an equation that seems to be dividing the RGB by var(--text-opacity), and, removing that seemed to fix the issue. I'm not entirely sure why it was an issue in the first place.

**Reply by FeivelRiodan** [reply]
sorry i'm too late but the color exists on the intellisense but no matter what i do, it won't appear. i know that this isn't a browser problem because i can definitely use the color with traditional css

**Reply by homocodian** [reply]
> sorry i'm too late but the color exists on the intellisense but no matter what i do, it won't appear. i know that this isn't a browser problem because i can definitely use the color with traditional css
> 

Check your tailwind config

**Comment by homocodian** [comment]
Can you send me the screenshot of your tailwind.config file.

**Comment by Jorge-Ortiz-M** [comment]
Am I doing it as it should?

![image](https://user-images.githubusercontent.com/104739183/175783284-b56d1690-0f18-40c2-9e3a-c9397b44e57e.png)

**Reply by homocodian** [reply]
You should use like "bg-backcolor" for background color and for text color use "text-backcolor", that should work

**Comment by Azalea9X** [comment]
I think so? That is what worked for me. I do not know about everything
else.

Regards,

Jacob Siegel

**Comment by darius00klokj** [comment]
Hey guys, 

Not sure why this does not work for me, it also does not work on the [Tailwind playground.](https://play.tailwindcss.com/)

im following these docs: https://tailwindcss.com/docs/configuration

<img width="1491" alt="Screenshot 2023-02-15 at 12 24 16" src="https://user-images.githubusercontent.com/5430672/219014643-922943e3-81d6-46ed-a505-b5bea1c3f366.png">

Can someone point out the obvious mistake im making

Thanks!

**Reply by Ekeu** [reply]
Hello @darius00klokj 

If you want to use that `sky` color, you must write `text-sky`.

`text-sky-900` is used when you have multiple variants. And in this case, to declare other variants of the `sky` color, the syntax is:

`
theme {
colors: {
'sky': {
100: "#..."
200: "#..."
900: "#..."
}
}
}
`

**Reply by ChillhopMusic** [reply]
Hi Ekeu, 

Thanks for your reply, it works now 👍 would be good if this can be specified in the docs. 

Also, it would be nice if we can default all weights to a colour. 

Cheers

**Reply by ygg-m** [reply]
> would be good if this can be specified in the docs.

It does [here](https://tailwindcss.com/docs/customizing-colors#color-object-syntax)

**Reply by walnut937** [reply]
wait you have configured the colors inside the extend {} , but it is shown in tutorial to include it in theme

**Reply by raycris** [reply]
For me work when I put backgroundColor then my custom colors

### BEFORE
![BEFORE](https://user-images.githubusercontent.com/45718089/226421874-aa311395-c5bf-4868-a9b4-fdb0c77404ad.jpg)


### AFTER
![after](https://user-images.githubusercontent.com/45718089/226421953-f7db069f-e0ad-481d-90c3-3f4042361fc4.jpg)

**Reply by raycris** [reply]
Also you need to build again you tailwind file. Well at least that works for me.

**Reply by madmenmarketingindia** [reply]
Just use in the extended. It is also written in documentation  ==> 
https://tailwindcss.com/docs/customizing-colors#adding-additional-colors

![colorProperty](https://github.com/tailwindlabs/tailwindcss/assets/136569132/8816f50b-3bb0-4954-8573-2625f215164c)
![image](https://github.com/tailwindlabs/tailwindcss/assets/136569132/dcc088de-5eb0-426c-957b-7497fb203208)

**Reply by homocodian** [reply]
Try puttings colors under theme instead of extend

**Comment by taajason** [comment]
I faced this issue when using `tailwind.config.ts` in nextjs, it worked when I changed `tailwind.config.ts` to `tailwind.config.js`, I found the reason was that the ts file was not compiled automatically js file, hope it helps.
